ttnn/cpp/ttnn/operations/data_movement/repeat/repeat.cpp

```diff
@@ -18,10 +21,18 @@ ttnn::Tensor RepeatOperation::invoke(
     const Shape & repeat_dims,
     const std::optional<MemoryConfig>& memory_config_arg) {
 
-    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    // special path for handling padded tiled tensors that are naively repeated
+    // *with padding* in the current implementation
+    std::optional<Tensor> formatted_input_tensor = std::nullopt;
+    if (input_tensor.get_layout() != Layout::ROW_MAJOR
+        && input_tensor.get_legacy_shape().without_padding() != input_tensor.get_legacy_shape()) {
+        formatted_input_tensor = ttnn::untilize(input_tensor);
+    }
+
+    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({formatted_input_tensor.has_value() ? *formatted_input_tensor : input_tensor}))};
     operation::launch_op(
         [repeat_dims, memory_config_arg] (const std::vector<Tensor>& input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors, const std::vector<std::optional<Tensor>>& optional_output_tensors) -> std::vector<Tensor> {
```

Comment:
can we pass single `input_tensor`? Not sure, whether we need to construct a vector. Seems like it will always contain single element :)

Reply:
I'm not sure to be honest; I was only adapting what was there before. I'll give it a shot :)

Reply:
Doesn't seem like the get_workers_for_op_output function has a singleton overload, so I can't do anything about this for now.